src/pages/Signin/Signin.js

```diff
@@ -0,0 +1,99 @@
+import {  Button, Flex, Heading, Input, Text } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { loginUser } from "../../features/authenticationSlice";
+
+const Signin = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [userCredentials, setUserCredentials] = useState({
+    username: "",
+    password: "",
+  });
+
+  const testUserSigninHandler = () => {
+    dispatch(
+      loginUser({
+        username: "adarshbalika",
+        password: "adarshBalika123",
+      })
+    );
+    navigate("/feedpage");
+  };
+
+  const signinHandler = () => {
+    if (userCredentials.username && userCredentials.password !== "") {
+      dispatch(loginUser(userCredentials));
+      navigate("/feedpage");
```

Comment:
you are navigating to the `feedpage` assuming without checking the auth state. It might be possible that userid or password is incorrect and user might want to retry signin.